### PR TITLE
Include select_options.properties in native-image

### LINKED ISSUE #825: Junit vs JUnit
Currently, the test framework options shown in the UI are `Junit`, `Spock`, and `Kotest`.

I think `JUnit` is correct, vs `Junit`.

### PR DESCRIPTION
**Pull request by ilopmar**
Fixes #825

### CODE REVIEW

File: starter-api/src/main/resources/META-INF/native-image/io.micronaut.starter.api/micronaut-starter-api/resource-config.json [hunk 1]
```diff
@@ -3,6 +3,7 @@
     {"pattern":"\\QMETA-INF/swagger\\E"},
     {"pattern":".*/swagger/micronaut-launch-.*yml"},
     {"pattern":"\\QMETA-INF/swagger/views/rapidoc/index.html\\E"},
-    {"pattern":"\\QMETA-INF/swagger/views/swagger-ui/index.html\\E"}
+    {"pattern":"\\QMETA-INF/swagger/views/swagger-ui/index.html\\E"},
+    {"pattern":"\\select_options.properties\\E"}
```
Comment (eahrold) [inline starter-api/src/main/resources/META-INF/native-image/io.micronaut.starter.api/micronaut-starter-api/resource-config.json]: How are you building your native-image?

When I run  (with `sdk use java 21.1.0.r11-grl`)

```
native-image --no-fallback -cp starter-web-netty-2.5.6-SNAPSHOT-all.jar
```

I get

```
[micronaut-starter:99711]    classlist:   1,989.95 ms,  0.94 GB
[micronaut-starter:99711]        (cap):   2,520.20 ms,  1.19 GB
[micronaut-starter:99711]        setup:   4,185.36 ms,  1.19 GB
[micronaut-starter:99711]     analysis:   7,650.95 ms,  1.70 GB
Fatal error:java.util.regex.PatternSyntaxException: Illegal/unsupported escape sequence near index 27
\select_options.properties\E
                                              ^
        at java.base/java.util.regex.Pattern.error(Pattern.java:2027)
        at java.base/java.util.regex.Pattern.escape(Pattern.java:2616)
...
```

Comment (ilopmar) [reply starter-api/src/main/resources/META-INF/native-image/io.micronaut.starter.api/micronaut-starter-api/resource-config.json]: Damn it! I missed the `Q`. @eahrold please try with 
`{"pattern":"\\Qselect_options.properties\\E"}` and if it works, please push the commit with the fix. Sorry about it

**Comment by ilopmar** [comment]
Thanks @eahrold :+1:

File: starter-api/src/main/resources/META-INF/native-image/io.micronaut.starter.api/micronaut-starter-api/resource-config.json (hunk 1 above)
Comment (eahrold) [reply starter-api/src/main/resources/META-INF/native-image/io.micronaut.starter.api/micronaut-starter-api/resource-config.json]: Pushed escape sequence fix, but it still doesn't look like it's picking up the resource for some reason.

http://localhost:8080/select-options

```
    "test": {
        "options": [
            {
                "description": "Junit",
                "name": "junit",
                "value": "JUNIT",
                "label": "Junit",
                "_links": {}
            },
```

**Comment by ilopmar** [comment]
I've pushed a commit that fixes the issue (I've tried it locally). The problem is that the properties needs to be declared as a bundle, not as a resource.

**Review by eahrold — APPROVED** [review]
@ilopmar Excellent.  Looks good to me, and I can verify it's working.

**Pull request merged**